src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,99 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFOULT_SIZE = 10;
+    private int size;
+    private Object[] values;
+
+    public ArrayList() {
+        values = new Object[DEFOULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (values.length == size) {
+            resize();
+        }
+        values[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index != size) {
+            checkIndex(index);
+        }
+        if (++size == values.length) {
+            resize();
+        }
 
+        System.arraycopy(values, index, values, index + 1, size - index);
+        values[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) values[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        values[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T remoteElement = (T) values[index];
+        System.arraycopy(values, index + 1, values, index, size - 1 - index);
+        size--;
+        return remoteElement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        T res;
```

Comment:
you can move it on 69 line: T res = (T) values[i]; 
and make your parameter res more informative like removedResult or something else